tracee-rules/signatures/rego/regosig/traceerego.go

```diff
@@ -43,7 +43,13 @@ func NewRegoSignature(regoCodes ...string) (types.Signature, error) {
 
 	var pkgName string
 	for _, regoCode := range regoCodes {
-		regoModuleName := strings.Split(re.FindString(regoCode), " ")[1]
+		var regoModuleName string
+		splittedName := strings.Split(re.FindString(regoCode), " ")
+		if len(splittedName) > 1 {
+			regoModuleName = splittedName[1]
+		} else {
```

Comment:
not sure if we should return here...
anyways we could here panic if there is no entry at index 1.

Reply:
Continue?

Reply:
well, not exactly: if the signatures imports a helper.rego then it will output weird logs or have an incorrect behaviour. That's why I would prefer to just return error.

Reply:
failing the entire process just because one rego policy was invalid doesn't feel right to me (think also about the security implication of that). Of course today this is what happens because we don't handle the error at all, but if we take the opportunity to fix it, let's choose the correct outcome.

I think continue is better:
- signature invalid: only the sig will be skipped
- tracee helper is invalid: not possible, we wrote the helpers so we can assume it's valid
- user-defined helper is invalid: only the user-provided policies will fail

^^ that sounds like a more reasonable fail model than always quit the process.

WDYT?

Reply:
what do you mean by security implication? the user of the library tries to load an incomplete signature, you shouldn't allow that. but sure, if you want we can have a continue statement.

Reply:
it's not just my decision, I genuinely ask what you think. by security implications I meant that someone can drop a malformed rego file in the rules dir and break tracee (if you will return an error here)

Reply:
I think continuing is fine as long as long as it's properly logged (regardless of if debug is set) or otherwise communicated to the user. If we do return, the error message should be expanded so the user knows which rego file was invalid.

Reply:
I believe returning is safer as a user could load a signatures that uses a helper file (for rego) and succeeding in loading (the function will report any error, but once the signature is loaded then strange error logs will be output. by returning we will be on the safe side. that's my opinion. 
Using a continue statement + log will not prevent from printing errors once the signature will not find its helper file.

Reply:
after a closer look (thanks @yanivagman) I realized that returning an error does exactly what I wanted to happen, since the caller (findRegoSigs) will do that continue.